Modules/_elementtree.c

```diff
@@ -2122,20 +2138,21 @@ elementiter_dealloc(ElementIterObject *it)
     Py_XDECREF(it->sought_tag);
     Py_XDECREF(it->root_element);
 
-    tp->tp_free(it);
+    tp->tp_free(op);
     Py_DECREF(tp);
 }
 
 static int
-elementiter_traverse(ElementIterObject *it, visitproc visit, void *arg)
+elementiter_traverse(PyObject *op, visitproc visit, void *arg)
 {
+    ElementIterObject *it = _ElementIter_CAST(op);
     Py_ssize_t i = it->parent_stack_used;
     while (i--)
         Py_VISIT(it->parent_stack[i].parent);
 
     Py_VISIT(it->root_element);
     Py_VISIT(it->sought_tag);
-    Py_VISIT(Py_TYPE(it));
+    Py_VISIT(Py_TYPE(op));
```

Comment:
*it* would be accepted, why not passing it to reduce the number of changes?

Reply:
I'm fine with both but since `Py_TYPE` expects a `PyObject *`, I prefer passing a "PyObject *" as well (just for semantics). By the way, Py_TYPE can be a macro which, without the Limited API, appears to cast it (_Py_Type also accepts a "PyObject *"):

```C
    #if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 < 0x030b0000
    #   define Py_TYPE(ob) _Py_TYPE(_PyObject_CAST(ob))
    #else
    #   define Py_TYPE(ob) _Py_TYPE(ob)
    #endif
```

But considering the rest of your review, I'll go for something that keeps the diff smaller.